src/NServiceBus.Metrics/RawData/RawDataReporter.cs

```diff
@@ -0,0 +1,91 @@
+﻿namespace NServiceBus.Metrics.RawData
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using Hosting;
+    using Logging;
+    using Routing;
+    using Support;
+    using Transport;
+
+    delegate void WriteOutput(ArraySegment<RingBuffer.Entry> entries, BinaryWriter outputWriter);
+
+    class RawDataReporter
+    {
+        readonly RingBuffer buffer;
+        readonly Action<ArraySegment<RingBuffer.Entry>> outputWriter;
+        readonly IDispatchMessages dispatcher;
+        readonly UnicastAddressTag destination;
+        readonly TransportTransaction transportTransaction = new TransportTransaction();
+        readonly Dictionary<string, string> headers = new Dictionary<string, string>();
+        readonly BinaryWriter writer;
+        readonly MemoryStream memoryStream;
+        readonly CancellationTokenSource cancellationTokenSource;
+        Task reporter;
+        static readonly TimeSpan delayTime = TimeSpan.FromSeconds(1);
+        static ILog log = LogManager.GetLogger<RawDataReporter>();
+        
+        public RawDataReporter(IDispatchMessages dispatcher, string destination, HostInformation hostInformation, RingBuffer buffer, string messageTypeName, string endpointName,
+            WriteOutput outputWriter)
+        {
+            this.buffer = buffer;
+            this.outputWriter = entries=>outputWriter(entries,writer);
+            this.dispatcher = dispatcher;
+            this.destination = new UnicastAddressTag(destination);
+
+            headers[Headers.OriginatingMachine] = RuntimeEnvironment.MachineName;
+            headers[Headers.OriginatingHostId] = hostInformation.HostId.ToString("N");
+            headers[Headers.EnclosedMessageTypes] = "NServiceBus.Metrics." + messageTypeName; 
+            headers[Headers.OriginatingEndpoint] = endpointName;
+
+            memoryStream = new MemoryStream();
```

Comment:
Nope. The `memoryStream` and the `writer` are reused for sending messages over and over again. I see, that `RawDataReporter` could be disposable but this would require to first `.Stop()` it and then `.Dispose()` it. Is this the scenario you mean?